.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.10
```

Comment:
@Chocobo1 No builds currently use `RC_1_2` HEAD. That comment is just for reference to explain that, in the current version of vcpkg, passing `--head` when installing libtorrent would pull in `RC_1_2` HEAD. As explained in https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491114697, these comments are meant to be adjusted as one bumps the vcpkg version.

Reply:
@glassez 
The libtorrent version is fixed to some git commit and requires regularly bumps to keep up with the latest development.
Are you fine with this? I know you already approved, so I'm just double checking.

Reply:
@Chocobo1 

>The libtorrent version is fixed to some git commit

That is possible, but currently it is fixed to a tag (`libtorrent-1.2.10`), not a commit. Though the case can be made to pin exclusively to commits, as tags are mutable (in the sense that it is possible to change the commit they point to) on the remote, commit hashes are not.

> and requires regularly bumps to keep up with the latest development.

There is no requirement to keep this regularly updated, though I think it is useful to bump whenever some important change lands in libtorrent that may have important implications in our CI, or when whenever we introduce patches that includes code of the form `if LIBTORRENT_VERSION >= 1.2.XX`, where `1.2.XXX` is greater than the current version in CI (yes, I will deal with libtorrent 2.0 later...).

Plus, being able to bump the version with just a single small change committed in a text file is very powerful, compared to having to manually preparing a custom build and hosting it on builds.shiki.hu.

Reply:
>@glassez
The libtorrent version is fixed to some git commit and requires regularly bumps to keep up with the latest development.
Are you fine with this? I know you already approved, so I'm just double checking.

I don't mind being sticked to a specific libtorrent version.
But as I've said many times, we need to stick to the versions of libraries/tools (of course, among the supported ones) that are least likely to be used by team members to make sure that we don't accidentally break their support. And since most of us prefer to use the fresh versions (the latest or close to the latest), in CI environment, we should use the minimum supported versions or the closest to them if possible.